normalizer_lib.py

```diff
@@ -1,205 +1,216 @@
 """Library of language-agnostic FST rewrite rules to normalize text."""
 
+import importlib
 import unicodedata
-from dataclasses import dataclass
 from typing import List
 from pynini import *
 from pynini.lib import byte, pynutil
-import config.zu as zu
 
-LANGUAGE = zu
-
-GRAPHEMES = LANGUAGE.GRAPHEMES
-
-INITIAL_PUNCTUATION = LANGUAGE.INITIAL_PUNCTUATION
-FINAL_PUNCTUATION = LANGUAGE.FINAL_PUNCTUATION
-OTHER_PUNCTUATION = union(r"\[", r"\]")
-PUNCTUATION = union(INITIAL_PUNCTUATION, FINAL_PUNCTUATION, OTHER_PUNCTUATION)
-NUMERALS = LANGUAGE.NUMERALS
-UNDERSCORE = acceptor("_")
-SIGMA_STAR = byte.BYTES.closure()
-
-
-@dataclass
-class Verbalizable:
-    """Contains verbalizable tokens."""
-
-    @staticmethod
-    def verbalizable():
+class NormalizerLib:
+    """Loads language-specific information for the text normalizer."""
+
+    def __init__(self, language):
+        self.language = importlib.import_module("config."+language)
+        self.graphemes = self.language.GRAPHEMES
+        self.initial_punctuation = self.language.INITIAL_PUNCTUATION
+        self.final_punctuation = self.language.FINAL_PUNCTUATION
+        self.other_punctuation = union(r"\[", r"\]")
+        self.punctuation = union(self.initial_punctuation,
+                                 self.final_punctuation,
+                                 self.other_punctuation)
+        self.non_grapheme_punctuation = difference(self.punctuation,
+                                                   self.graphemes)
+        self.numerals = self.language.NUMERALS
+        self.sigma_star = byte.BYTES.closure()
+
+        self.dummy_rewrite = cdrewrite(cross("", ""), "", "", self.sigma_star)
+        try:
+            self.language_specific_preprocessing = (
+                self.language.LANGUAGE_SPECIFIC_PREPROCESSING)
+        except Exception:
+            self.language_specific_preprocessing = self.dummy_rewrite
+        self.remove_extra_whitespace = cdrewrite(
+            pynutil.delete(byte.SPACE),
+            "",
+            union(byte.SPACE, "[EOS]"),
+            self.sigma_star)
+        self.detach_leading_punctuation = cdrewrite(
+            pynutil.insert(" "),
+            union("[BOS]", byte.SPACE) + self.non_grapheme_punctuation.plus,
+            union(self.non_grapheme_punctuation,
+                  self.graphemes,
+                  self.numerals),
+            self.sigma_star)
+        self.detach_trailing_punctuation = cdrewrite(
+            pynutil.insert(" "),
+            union(self.non_grapheme_punctuation,
+                  self.graphemes,
+                  self.numerals,
+                  "/"),
+            self.punctuation.plus + union("[EOS]", byte.SPACE),
+            self.sigma_star)
+        self.delete_freestanding_punctuation = cdrewrite(
+            pynutil.delete(self.punctuation),
+            union("[BOS]", byte.SPACE),
+            union("[EOS]", byte.SPACE),
+            self.sigma_star)
+
+
+    def verbalizable(self):
```

Comment:
Can we add the returns annotation to this line?

def verbalizable(self) -> Fst:

Reply:
Done!